tree/dataframe/src/RDataFrame.cxx

```diff
@@ -1232,26 +1232,56 @@ Notice how we created one `double` variable for each processing slot and later m
 
 
 \anchor friends
-### Friend trees
-Friend TTrees are supported by RDataFrame.
-Friend TTrees with a TTreeIndex are supported starting from ROOT v6.24.
+### Dataset joins with friend trees
 
-To use friend trees in RDataFrame, it is necessary to add the friends directly to
-the tree and instantiate an RDataFrame with the main tree:
+Vertically concatenating multiple trees that have the same columns (creating a logical dataset with the same columns and
+more rows) is trivial in RDataFrame: just pass all tree and file names to RDataFrame's constructor, or create a TChain
+out of the desired trees and pass that to RDataFrame.
+
+Horizontal concatenations of trees or chains (creating a logical dataset with the same number of rows and the union of the
+columns of multiple trees) leverages TTree's "friend" mechanism.
+
+Simple joins of trees that do not have the same number of rows are also possible with indexed friend trees (see below).
+
+To use friend trees in RDataFrame, set up trees with the appropriate relationships and then instantiate an RDataFrame
+with the main tree:
 
 ~~~{.cpp}
-TTree t([...]);
-TTree ft([...]);
-t.AddFriend(&ft, "myFriend");
+TTree main([...]);
+TTree friend([...]);
+main.AddFriend(&friend, "myFriend");
 
-RDataFrame d(t);
-auto f = d.Filter("myFriend.MyCol == 42");
+RDataFrame df(main);
+auto df2 = df.Filter("myFriend.MyCol == 42");
 ~~~
 
-Columns coming from the friend trees can be referred to by their full name, like in the example above,
+The same applies for TChains. Columns coming from the friend trees can be referred to by their full name, like in the example above,
 or the friend tree name can be omitted in case the column name is not ambiguous (e.g. "MyCol" could be used instead of
-      "myFriend.MyCol" in the example above).
+"myFriend.MyCol" in the example above if there is no column "MyCol" in the main tree).
+
+\note A common source of confusion is that trees that are written out from a multi-thread Snapshot() call will have their
```

Comment:
Maybe a warning could be better in this case since it's a source of confusion, we want people to see this paragraph as easily as possible

Reply:
notes are quite visible and a warning seems to harsh to me: this is a clarification on a concept, not a "don't use this" or "you are probably doing something wrong" kind of thing

![image](https://github.com/root-project/root/assets/10999034/4f9f2b46-c9ff-43c4-a017-2b27509875c1)